dynamic_programming/house_robber.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file
+ * @brief Implementation of [House Robber
+ * Problem](https://labuladong.gitbook.io/algo-en/i.-dynamic-programming/houserobber)
+ * algorithm
+ * @details
+ * Solution of House robber problem uses a dynamic programming concept that
+ * works in \f$O(n)\f$ time and works in \f$O(1)\f$ space.
+ * @author [Swastika Gupta](https://github.com/Swastyy)
+ */
+
+#include <cassert>   /// for assert
+#include <climits>   /// for std::max
+#include <iostream>  /// for io operations
+#include <vector>    /// for std::vector
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @namespace house_robber
+ * @brief Functions for House Robber algorithm
+ */
+namespace house_robber {
+/**
+ * @brief The main function implements house robber algorithm using dynamic
+ * programming
+ * @param money array containing money in the ith house
+ * @param n size of array
+ * @returns maximum amount of money that can be robbed
+ */
+std::uint64_t houseRobber(const std::vector<int> &money, int n) {
```

Comment:
`n` is not being modified here, so, we should make it a `const` and pass it by reference.
```suggestion
std::uint64_t houseRobber(const std::vector<int> &money, const int &n) {
```